service.js

```diff
@@ -0,0 +1,86 @@
+import request from 'request';
+import uuid from 'node-uuid';
+import promisify from './promisify';
+import ConsultantError from './consultant-error';
+
+const deregister = async (host, instance) => {
+	const req = {
+		url : `${host}/v1/agent/service/deregister/${instance.instance}`,
+		method : 'DELETE',
```

Comment:
Just because you can, doesn't mean you should. Additionally I remember something about nginx liking to retry GETs, which might not be what you want here.